scripts/dev/configure_container_auth.sh

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bash
+
+set -Eeou pipefail
+test "${MDB_BASH_DEBUG:-0}" -eq 1 && set -x
+
+source scripts/dev/set_env_context.sh
+source scripts/funcs/checks
+source scripts/funcs/printing
+source scripts/funcs/kubernetes
+
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME-"docker"}"
+
+setup_validate_container_runtime() {
+  case "${CONTAINER_RUNTIME}" in
+    "podman")
+      if ! command -v podman &> /dev/null; then
+        echo "Error: Podman is not available but was specified"
+        exit 1
+      fi
+      USE_SUDO=true
+      CONFIG_PATH="/root/.config/containers/auth.json"
+      echo "Using Podman for container authentication (sudo mode)"
+      ;;
+    "docker")
+      if ! command -v docker &> /dev/null; then
+        echo "Error: Docker is not available but was specified"
+        exit 1
+      fi
+      USE_SUDO=false
+      CONFIG_PATH="${HOME}/.docker/config.json"
+      echo "Using Docker for container authentication"
+      ;;
+    *)
+      echo "Error: Invalid container runtime '${CONTAINER_RUNTIME}'. Must be 'docker' or 'podman'"
+      exit 1
+      ;;
+  esac
+
+  if [[ "${USE_SUDO}" == "true" ]]; then
+    sudo mkdir -p "$(dirname "${CONFIG_PATH}")"
+  else
+    mkdir -p "$(dirname "${CONFIG_PATH}")"
+  fi
+}
+
+# Wrapper function to execute commands with or without sudo
+exec_cmd() {
+  if [[ "${USE_SUDO}" == "true" ]]; then
+    sudo env PATH="${PATH}" "$@"
+  else
+    "$@"
+  fi
+}
+
+# Wrapper function to read files with or without sudo
+read_file() {
+  local file="$1"
+  if [[ "${USE_SUDO}" == "true" ]]; then
+    sudo cat "${file}"
+  else
+    cat "${file}"
+  fi
+}
+
+# Wrapper function to write files with or without sudo
+write_file() {
+  local content="$1"
+  local file="$2"
+  if [[ "${USE_SUDO}" == "true" ]]; then
+    echo "${content}" | sudo tee "${file}" > /dev/null
+  else
+    echo "${content}" > "${file}"
+  fi
+}
+
+remove_element() {
+  local config_option="$1"
+  local tmpfile
+  tmpfile=$(mktemp)
+
+  if [[ ! -f "${CONFIG_PATH}" ]]; then
+    write_file '{}' "${CONFIG_PATH}"
+  fi
+
+  exec_cmd jq 'del(.'"${config_option}"')' "${CONFIG_PATH}" > "${tmpfile}"
+  exec_cmd cp "${tmpfile}" "${CONFIG_PATH}"
+  rm "${tmpfile}"
+}
+
+container_login() {
```

Comment:
nit:
should we call this `registry_login` instead of `container_login`? `contianer_login` gives the feel that we are logging into a container.